src/server/game/Server/Packets/PartyPackets.cpp

```diff
@@ -0,0 +1,354 @@
+    _worldPacket.WriteBits(InviterRealmNameNormalized.length(), 8);
+    _worldPacket.FlushBits();
+    _worldPacket.WriteString(InviterRealmNameActual);
+    _worldPacket.WriteString(InviterRealmNameNormalized);
+
+    _worldPacket << uint32(ProposedRoles);
+    _worldPacket << uint32(LfgSlotsCount);
+    _worldPacket << uint32(LfgCompletedMask);
+
+    _worldPacket.WriteString(InviterName);
+
+    for (uint32 lfgSlot : LfgSlots)
+        _worldPacket << lfgSlot;
+
+    return &_worldPacket;
+}
+
+WorldPacket const* WorldPackets::Party::PartyCommandResult::Write()
+{
+    _worldPacket.WriteBits(Name.length(), 9);
+    _worldPacket.WriteBits(Command, 4);
+    _worldPacket.WriteBits(Result, 6);
+    _worldPacket.FlushBits();
+    _worldPacket << uint32(ResultData);
+    _worldPacket << ResultGUID;
+    _worldPacket.WriteString(Name);
+
+    return &_worldPacket;
+}
+
+WorldPacket const* WorldPackets::Party::RoleChangedInform::Write()
+{
+    _worldPacket << PartyIndex;
+    _worldPacket << From;
+    _worldPacket << ChangedUnit;
+    _worldPacket << uint32(OldRole);
+    _worldPacket << uint32(NewRole);
+
+    return &_worldPacket;
+}
+
+WorldPacket const* WorldPackets::Party::PartyUpdate::Write()
+{
+    _worldPacket << uint8(PartyFlags);
+    _worldPacket << uint8(PartyIndex);
+    _worldPacket << uint8(PartyType);
+
+    _worldPacket << uint32(MyIndex);
+    _worldPacket << PartyGUID;
+    _worldPacket << uint32(SequenceNum);
+    _worldPacket << LeaderGUID;
+
+     _worldPacket << uint32(PlayerList.size()); //PlayerListCount
+    for (PlayerInfo const& player : PlayerList)
+    {
+        _worldPacket.WriteBits(player.Name.length(), 6);
+        _worldPacket.FlushBits();
+
+        _worldPacket << player.Guid;
+
+        _worldPacket << uint8(player.Connected);
+        _worldPacket << uint8(player.Subgroup);
+        _worldPacket << uint8(player.Flags);
+        _worldPacket << uint8(player.RolesAssigned);
+        _worldPacket << uint8(player.UnkByte);
+
+        _worldPacket.WriteString(player.Name);
+    }
+
+    _worldPacket.WriteBit(HasLfgInfo);
+    _worldPacket.WriteBit(HasLootSettings);
+    _worldPacket.WriteBit(HasDifficultySettings);
+    _worldPacket.FlushBits();
+
+    if (HasLfgInfo)
+    {
+        _worldPacket << MyLfgFlags;
+        _worldPacket << LfgSlot;
+        _worldPacket << MyLfgRandomSlot;
+        _worldPacket << MyLfgPartialClear;
+        _worldPacket << float(MyLfgGearDiff);
+        _worldPacket << MyLfgStrangerCount;
+        _worldPacket << MyLfgKickVoteCount;
+        _worldPacket << LfgBootCount;
+
+        _worldPacket.WriteBit(LfgAborted);
+        _worldPacket.WriteBit(MyLfgFirstReward);
+        _worldPacket.FlushBits();
+    }
+
+    if (HasLootSettings)
+    {
+        _worldPacket << LootMethod;
+        _worldPacket << LootMaster;
+        _worldPacket << LootThreshold;
+    }
+
+    if (HasDifficultySettings)
+    {
+        _worldPacket << UnkInt4;
```

Comment:
@xerkoss May I inquire what "legacy" refers to? I only know of legacy achievements which are old achievements that are now deprecated.